Tools/cases_generator/analyzer.py

```diff
@@ -1079,6 +1081,35 @@ def add_instruction(name: str) -> None:
     return instmap, len(no_arg), min_instrumented
 
 
+def is_instruction_size_macro(token: lexer.Token) -> bool:
+    return token.kind == "IDENTIFIER" and token.text == "INSTRUCTION_SIZE"
+
+
+def contains_instruction_size_macro(uop: Uop) -> bool:
+    return any(is_instruction_size_macro(token) for token in uop.body)
+
+
+def get_instruction_size_for_uop(instructions: dict[str, Instruction], uop: Uop) -> int:
+    """Return the size of the instruction that contains the given uop.
+
+    If there is more than one instruction that contains the uop,
+    ensure that they all have the same size.
+    """
+    size = None
+    for inst in instructions.values():
+        if uop in inst.parts:
+            if size is None:
+                size = inst.size
+            assert size == inst.size, (
```

Comment:
This should not be an assertion. This is an input error, not a logical one.
```suggestion
            raise analysis_error(
```